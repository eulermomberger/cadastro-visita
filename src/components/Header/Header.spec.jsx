import renderer from 'react-test-renderer';

import { Header } from '.';

jest.mock('firebase/firestore/lite', () => {
  return {
    collection() {
      return null;
    },
    getDocs() {
      return {
        docs: [{
          data() {
            return {
              name: 'Luis',
              rg: '0192839382',
              updated_at: {
                toDate() { return new Date('2023-09-24'); },
              },
            };
          },
          uuid: 'abcd123',
        }],
      };
    },
    orderBy() {},
    query() {},
    where() {},
    getFirestore() {},
  };
});

describe('Header component', () => {
  it('Should renders correctly', () => {
    const testRenderer = renderer.create(
      <Header
        openModal={() => {}}
        setVisitors={() => {}}
      />
    );

    try {
      expect(testRenderer.toJSON()).toMatchSnapshot();
    } finally {
      testRenderer.unmount();
    }
  });
});
