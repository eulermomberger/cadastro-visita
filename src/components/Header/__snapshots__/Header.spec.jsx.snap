// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component Should renders correctly 1`] = `
<div
  className="header-container"
>
  <img
    alt="Logo da aplicação - Três prédios e ao lado o título cadastro visita"
    src="/logo.svg"
  />
  <div
    className="header-search"
  >
    <div
      className="search-container"
    >
      <input
        className="search-input"
        onChange={[Function]}
        placeholder="Buscar cadastro"
      />
      <svg
        color="#FFF"
        fill="currentColor"
        height="24px"
        stroke="currentColor"
        strokeWidth="0"
        style={
          {
            "color": "#FFF",
          }
        }
        viewBox="0 0 24 24"
        width="24px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M15.5 14h-.79l-.28-.27A6.471 6.471 0 0016 9.5 6.5 6.5 0 109.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
        />
      </svg>
    </div>
  </div>
  <div
    className="popup-button-container"
  >
    <button
      className="button-container"
      onClick={[Function]}
      style={
        {
          "backgroundColor": "#6933FF",
        }
      }
    >
      Cadastrar novo visitante
    </button>
  </div>
</div>
`;
